activerecord/lib/active_record/relation/from_clause.rb

```diff
@@ -25,7 +25,7 @@ def empty?
       end
 
       def self.empty
-        new(nil, nil)
+        @empty ||= new(nil, nil).freeze
```

Comment:
I'd prefer we better express that the object is meant to be immutable by documenting it as such, rather than adding code changes which will cause the uninitiated pain when they want to add a method that needs to memoize.